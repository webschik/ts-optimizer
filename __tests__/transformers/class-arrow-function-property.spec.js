const ts = require('typescript');
const {createClassArrowFunctionPropertyTransformerFactory} = require('../../src/transformers/class-arrow-function-property');

describe('Transformers', () => {
    describe('ClassArrowFunctionProperty', () => {
        const printer = ts.createPrinter();

        it('should convert class arrow function to bound methods', () => {
            const inputFile = ts.createSourceFile(
                'test.ts',
                `
                    class Button1 {
                        private fn1 = (a, b) => {};
                        private fn1 = (a, b, c) => {};
                        protected fn2 = () => {};
                        fn3 = () => {
                          return 2;
                        } 
                        
                        render () {
                            return <button />
                        }
                    }
                    
                    class Button2 extends React.Component {
                        private fn1 = (a, b) => {};
                        private fn1 = (a, b, c) => {};
                        protected fn2 = () => {};
                        fn3 = () => {
                          return 2;
                        }  
                        
                        render () {
                            return <button />
                        }
                    }
                    
                    class Button3 extends React.Component {
                        constructor (props) {
                            super(props);
                            this.state = {};
                        }
                    
                        private fn1 = (a, b) => {};
                        private fn1 = (a, b, c) => {};
                        protected fn2 = () => {};
                        fn3 = () => {
                          return 2;
                        } 
                        
                        render () {
                            return <button />
                        } 
                    }
                `,
                ts.ScriptTarget.ESNext,
                true,
                ts.ScriptKind.TSX
            );
            const result = ts.transform(inputFile, [
                createClassArrowFunctionPropertyTransformerFactory()
            ]);
            const [outputFile] = result.transformed;

            expect(outputFile).toBeDefined();
            const outputFileText = printer.printFile(outputFile);

            expect(printer.printFile(inputFile)).not.toBe(outputFileText);
            expect(outputFileText).toMatchSnapshot();
        });
    });
});