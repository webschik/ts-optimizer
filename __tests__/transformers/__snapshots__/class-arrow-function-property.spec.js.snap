// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transformers ClassArrowFunctionProperty should convert class arrow function to bound methods 1`] = `
"class Button1 {
    constructor() { this.fn1 = this.fn1.bind(this); this.fn2 = this.fn2.bind(this); this.fn3 = this.fn3.bind(this); }
    fn1(a, b) { }
    fn2() { }
    fn3() {
        return 2;
    }
    render() {
        return <button />;
    }
}
class Button2 extends React.Component {
    constructor(...args) { super(...args); this.fn1 = this.fn1.bind(this); this.fn2 = this.fn2.bind(this); this.fn3 = this.fn3.bind(this); }
    fn1(a, b) { }
    fn2() { }
    fn3() {
        return 2;
    }
    render() {
        return <button />;
    }
}
class Button3 extends React.Component {
    constructor(props) {
        super(props);
        this.state = {};
        this.fn1 = this.fn1.bind(this);
        this.fn2 = this.fn2.bind(this);
        this.fn3 = this.fn3.bind(this);
    }
    fn1(a, b) { }
    fn2() { }
    fn3() { return 2; }
    render() {
        return <button />;
    }
}
"
`;
